Engine/AlgorithmManager.cs

```diff
@@ -884,13 +927,31 @@ private static void HandleDelistedSymbols(IAlgorithm algorithm, Delistings newDe
                 // submit an order to liquidate on market close
                 if (delisting.Type == DelistingType.Warning)
                 {
-                    delistings.Add(delisting);
+	                SubmitOrderRequest request;
+
+	                delistings.Add(delisting);
                     Log.Trace("AlgorithmManager.Run(): Security delisting warning: " + delisting.Symbol.ID);
                     var security = algorithm.Securities[delisting.Symbol];
                     if (security.Holdings.Quantity == 0) continue;
-                    var submitOrderRequest = new SubmitOrderRequest(OrderType.MarketOnClose, security.Type, security.Symbol,
-                        -security.Holdings.Quantity, 0, 0, algorithm.UtcTime, "Liquidate from delisting");
-                    var ticket = algorithm.Transactions.ProcessRequest(submitOrderRequest);
+
+                    if (security.Type == SecurityType.Option)
```

Comment:
If statements like this make me think we're missing something with the modelling. Ideally the options would have its own model that performs special behavior and other securities would do the default behavior.

Maybe this makes sense as `ISecurityPortfolioModel.ProcessDelisting` or something similar